src/Security/MemberAuthenticator/ChangePasswordHandler.php

```diff
@@ -158,8 +158,10 @@ protected function setSessionToken($member, $token)
             Injector::inst()->get(IdentityStore::class)->logOut();
         }
 
-        // Store the hash for the change password form. Will be unset after reload within the ChangePasswordForm.
-        $this->getRequest()->getSession()->set('AutoLoginHash', $member->encryptWithUserSettings($token));
+        // Store the token for the change password form. Will be unset after reload within the ChangePasswordForm.
+        $session = $this->getRequest()->getSession();
+        $session->set('AutoLoginHash', $token);
+        $session->set('PasswordResetMemberID', $member->ID);
```

Comment:
The "Hash" is now a token - arguably the randomness of the token is reduced when "hashed with user info". Also now it's tracking the ID of the member in the session too, meaning that we can fetch the member and their reset tokens rather than scanning for all members.